lib/sycamore/sycamore/llms/prompts/prompts.py

```diff
@@ -45,8 +45,9 @@ class RenderedPrompt:
     response_format: ResponseFormat = None
 
     def token_count(self, tokenizer: Tokenizer) -> int:
-        total_text = " ".join(m.content for m in self.messages)
-        return len(tokenizer.tokenize(total_text))
+        if len(self.messages) == 0:
```

Comment:
If you add @cache onto the tokenize method in the tokenizer class I think it will auto-cache this so repeated calls are fine. I don't know if you still have the repeated calls.